udata/core/dataset/rdf.py

```diff
@@ -200,16 +201,25 @@ def dataset_to_rdf(dataset, graph=None):
     # unless there is already an upstream URI
     id = dataset_to_graph_id(dataset)
 
+    graph = graph or Graph(namespace_manager=namespace_manager)
+    d = graph.resource(id)
+
     # Expose upstream identifier if present
     if dataset.harvest and dataset.harvest.dct_identifier:
-        identifier = dataset.harvest.dct_identifier
+        d.set(DCT.identifier, Literal(dataset.harvest.dct_identifier))
+        alternate_identifier = URIRef(
```

Comment:
Shouldn't the identifier be a literal? because else it may be a resource described in the graph.

Reply:
I'm confused, didn't we decide to use [URIRef for the `endpointURL`](https://github.com/opendatateam/udata/pull/3179/files#r1821117993)? Is this any different?

I'm happy to change it, I would just love to understand the logics of when to use a Literal (simple string, right?) or a URIRef (I thought `ref` in `URIRef` meant "reference to another resource described in the graph", but then we used it for the `endpointURL` which is just a URL, not another resource described in the graph).

Reply:
I'm not 100% sure about the logic either.
I think that endpointURL is an identified resource on the web whereas an identifier is an identifying string?
However, we can base our decision on:
* the spec, which shows [endpoint URL](https://www.w3.org/TR/vocab-dcat-3/#Property:data_service_endpoint_url) in range Resource and [identifier having a notation](https://semiceu.github.io/DCAT-AP/releases/3.0.0/#Identifier) in range Literal
* usage in real life examples, ex: [this dataset on data.europa.eu](https://data.europa.eu/data/datasets/34fc887b-be29-47bf-a2b3-67428d47d434?locale=en) (with both adms:identifier and dcat:endpointUrl)